lib/chef/provider/package/windows/registry_uninstall_entry.rb

```diff
@@ -0,0 +1,78 @@
+#
+# Author:: Seth Chisamore (<schisamo@chef.io>)
+# Author:: Matt Wrock <matt@mattwrock.com>
+# Copyright:: Copyright (c) 2011, 2015 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'win32/registry' if (RUBY_PLATFORM =~ /mswin|mingw32|windows/)
+
+class Chef
+  class Provider
+    class Package
+      class Windows
+        class RegistryUninstallEntry
+
+          def self.find_entries(package_name)
+            entries = []
+            [
+              [::Win32::Registry::HKEY_LOCAL_MACHINE, (::Win32::Registry::Constants::KEY_READ | 0x0100)],
+              [::Win32::Registry::HKEY_LOCAL_MACHINE, (::Win32::Registry::Constants::KEY_READ | 0x0200)],
+              [::Win32::Registry::HKEY_CURRENT_USER]
+            ].each do |hkey|
+              desired = hkey.length > 1 ? hkey[1] : ::Win32::Registry::Constants::KEY_READ
+              begin
+                ::Win32::Registry.open(hkey[0], UNINSTALL_SUBKEY, desired) do |reg|
+                  reg.each_key do |key, _wtime|
+                    begin
+                      k = reg.open(key, desired)
+                      display_name = k['DisplayName'] rescue nil
+                      if display_name == package_name
+                        entries.push(RegistryUninstallEntry.new(hkey, key, k))
+                      end
+                    rescue ::Win32::Registry::Error
+                    end
+                  end
+                end
+              rescue ::Win32::Registry::Error
+              end
+            end
+            entries
+          end
+
+          def initialize(hive, key, registry_data)
+            @hive = hive
+            @key = key
+            @data = registry_data
+            @display_name = registry_data['DisplayName']
+            @display_version = registry_data['DisplayVersion'] rescue nil
```

Comment:
I'm not a fan of rescuing all the exceptions. I'm guessing in this case, you're expecting a registry error if the thing you're looking for cant be found? I think you should call a function like `maybe_registry_value` that only catches that error. I would still log to debug if you cant find it.